WireDomain/Sources/WireDomain/Repositories/User/UserRepository.swift

```diff
@@ -89,6 +103,67 @@ public final class UserRepository: UserRepositoryProtocol {
         }
     }
 
+    public func addUserClient(_ userClient: WireAPI.UserClient) async throws {
+        await context.perform { [context] in
+            let localUserClient: (client: WireDataModel.UserClient, isNew: Bool) = {
+                if let existingClient = UserClient.fetchExistingUserClient(
+                    with: userClient.id,
+                    in: context
+                ) {
+                    return (client: existingClient, isNew: false)
+                } else {
+                    let newClient = UserClient.insertNewObject(in: context)
+                    return (client: newClient, isNew: true)
+                }
+            }()
```

Comment:
I wonder if it would be clearer to have one function for inserting and one for updating on the repository and call that from the processor.

Or just rename this method to creatorupdate
What do you think ?

Reply:
I think it's a good idea to have two separate methods as you describe, one for fetch or create, and one for updating metadata. This is a pattern that we can apply to other repositories too.

Reply:
makes sense, I did the fixes + added some UTs, thx 👍